### PR TITLE
Replaced static methods and Added tests

### PR DESCRIPTION
**Pull request by abassawo**
Created a method in NameContestPresenter
and DescribeContestPresenter for checking if string is empty.

Method uses String.trim() to account for white space  - "     "

### CODE REVIEW

**Comment by gsintrepid** [comment]
@abassawo is this PR ready for review? (besides needing rebase)

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/describecontest/DescribeContestPresenter.java [hunk 1]
```diff
@@ -40,9 +39,14 @@ public void onNextValidated() {
         view.setNextEnabled(true);
     }
 
+    boolean isEmpty(CharSequence sequence) {
+        String sequenceString = String.valueOf(sequence);
+        return sequenceString.trim().isEmpty();
```
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/describecontest/DescribeContestPresenter.java]: I understand you did this to be able to test without a static Utils method... but adding this method in multiple presenters makes me confused.
I don't think you need `trim()` eihter; to make it simple I'd just add to `onTextChanged`:
```
if (newDescription.toString().isEmpty()) {
...
}
```

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenter.java [hunk 2]
```diff
@@ -42,9 +41,14 @@ public void onNextValidated() {
...
 
+    boolean isEmpty(CharSequence sequence) {
+        String sequenceString = String.valueOf(sequence);
+        return sequenceString.trim().isEmpty();
+    }
+
     @Override
     public void onTextChanged(CharSequence newName) {
-        if (TextUtils.isEmpty(newName)) {
+        if (isEmpty(newName)) {
```
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenter.java]: [Same comment]

**Comment by abassawo** [comment]
@gsintrepid Yeah those images snuck their way in during the last rebase. I'll try to clear that from this pr. Also, I used trim because the adverse result was occurring when entering empty text with just white space like 
"       ".  It occurs both with TextUtils.isEmpty() and String.isEmpty() interestingly enough.

File: app/src/test/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenterTest.java [hunk 3]
```diff
@@ -18,13 +21,14 @@
...
     Contest.Builder mockContestBuilder;
-    private NameContestContract.Presenter nameContestPresenter;
+    private NameContestPresenter nameContestPresenter;
 
     @Before
     public void setup(){
         MockitoAnnotations.initMocks(this);
-        nameContestPresenter = new NameContestPresenter(mockView, TestPresenterConfiguration.createTestConfiguration(),
-                                                        mockContestBuilder);
+        nameContestPresenter = spy(new NameContestPresenter(mockView,
```
Comment (kentonguan) [inline app/src/test/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenterTest.java]: any reason you're using spy here?

Comment (abassawo) [inline app/src/test/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenterTest.java]: I was using spy before when trying to find a workaround to test the static methods. I'm going to revert this though.

**Pull request closed without merging**